### PR TITLE
Separate 2d povray output from the general case.

### PR DESCRIPTION
**Pull request by bangerth**
The underlying cause of #13052 is that we compile the `DataOutBase::write_povray()` function for 1d, 2d, and 3d, even though it only does something useful in 2d (the 1d and 3d cases will throw an exception). This is wasteful, and also causes warnings.

This patch moves the actual implementation into an anonymous namespace, and specializes a function for the 2d case that we then call from the general template. The general template in the anonymous namespace is simply not implemented.

/rebuild

### CODE REVIEW

File: source/base/data_out_base.cc [hunk 1]
```diff
@@ -4068,330 +4068,364 @@ namespace DataOutBase
...
 #else
-    if (patches.size() == 0)
-      return;
+      if (patches.size() == 0)
+        return;
 #endif
-    Assert(dim == 2,
-           ExcNotImplemented()); // only for 2-D surfaces on a 2-D plane
-    Assert(spacedim == 2, ExcNotImplemented());
+      constexpr int dim      = 2;
```
Comment (Rombur) [inline source/base/data_out_base.cc:4107]: CI complains that this not used

**Review by Rombur — APPROVED** [review]
Looks good but you need to fix the warning from the CI

File: source/base/data_out_base.cc (hunk 1 above)
Comment (bangerth) [reply source/base/data_out_base.cc]: That's aggravating because it is used below in line 4175 :-( I'm pushing a patch for it.

Comment (Rombur) [reply source/base/data_out_base.cc]: Yes, it is used but only in debug mode.

**Pull request merged**